src/app/tests/TestDefaultEnhancedSetupFlowProvider.cpp

```diff
@@ -0,0 +1,309 @@
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#define CHIP_CONFIG_TC_REQUIRED 1                          // Enable conditional feature for testing
+#define CHIP_CONFIG_TC_REQUIRED_ACKNOWLEDGEMENTS 0         // Dummy define to satisfy #error check
+#define CHIP_CONFIG_TC_REQUIRED_ACKNOWLEDGEMENTS_VERSION 0 // Dummy define to satisfy #error check
+
+#include "app/server/DefaultEnhancedSetupFlowProvider.h"
+
+#include <lib/core/CHIPError.h>
+#include <pw_unit_test/framework.h>
+
+class FakeTermsAndConditionsProvider : public chip::app::TermsAndConditionsProvider
+{
+public:
+    FakeTermsAndConditionsProvider(uint16_t inAcceptedAcknowledgements, uint16_t inAcceptedAcknowledgementsVersion,
+                                   uint16_t inRequiredAcknowledgements, uint16_t inRequiredAcknowledgementsVersion) :
+        mAcceptedAcknowledgements(inAcceptedAcknowledgements),
+        mAcceptedAcknowledgementsVersion(inAcceptedAcknowledgementsVersion), mRequiredAcknowledgements(inRequiredAcknowledgements),
+        mRequiredAcknowledgementsVersion(inRequiredAcknowledgementsVersion)
+    {}
+
+    CHIP_ERROR GetAcceptance(uint16_t & outAcknowledgements, uint16_t & outAcknowledgementsVersion) const override
+    {
+        outAcknowledgements        = mAcceptedAcknowledgements;
+        outAcknowledgementsVersion = mAcceptedAcknowledgementsVersion;
+        return CHIP_NO_ERROR;
+    }
+
+    CHIP_ERROR GetRequirements(uint16_t & outAcknowledgements, uint16_t & outAcknowledgementsVersion) const override
+    {
+        outAcknowledgements        = mRequiredAcknowledgements;
+        outAcknowledgementsVersion = mRequiredAcknowledgementsVersion;
+        return CHIP_NO_ERROR;
+    }
+
+    CHIP_ERROR SetAcceptance(uint16_t inAcknowledgements, uint16_t inAcknowledgementsVersion) override
+    {
+        mAcceptedAcknowledgements        = inAcknowledgements;
+        mAcceptedAcknowledgementsVersion = inAcknowledgementsVersion;
+        return CHIP_NO_ERROR;
+    }
+
+    CHIP_ERROR ClearAcceptance() override
+    {
+        mAcceptedAcknowledgements        = 0;
+        mAcceptedAcknowledgementsVersion = 0;
+        return CHIP_NO_ERROR;
+    }
+
+private:
+    uint16_t mAcceptedAcknowledgements;
+    uint16_t mAcceptedAcknowledgementsVersion;
+    uint16_t mRequiredAcknowledgements;
+    uint16_t mRequiredAcknowledgementsVersion;
+};
+
+TEST(DefaultEnhancedSetupFlowProvider, TestNoAcceptanceRequiredCheckAcknowledgementsAcceptedSuccess)
+{
+    CHIP_ERROR err;
+    bool hasTermsBeenAccepted;
+
+    FakeTermsAndConditionsProvider tncProvider(0, 0, 0, 0);
+    chip::app::DefaultEnhancedSetupFlowProvider esfProvider;
+
+    err = esfProvider.Init(&tncProvider);
+    EXPECT_TRUE(CHIP_NO_ERROR == err);
```

Comment:
Replace all of the `EXPECT_TRUE(CHIP_NO_ERROR == err)` with `EXPECT_EQ(err, CHIP_NO_ERROR)` --> will allow seeing the value of error on test values